File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def divya(self, root, s, total_sum):
        if root is None:
            return
        if root.left is None and root.right is None:
            total_sum[0] += (s * 10 + root.val)
            return
        self.divya(root.left, s * 10 + root.val, total_sum)
        self.divya(root.right, s * 10 + root.val, total_sum)
        
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        total_sum = [0]
        self.divya(root, 0, total_sum)
        return total_sum[0]

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int sum;
    public void divya(TreeNode root,int s)
    {
        if(root==null)
        {
            return;
        }
        if(root.left==null && root.right==null)
        {
            sum+=(s*10+root.val);
            return;
        }
        divya(root.left,s*10+root.val);
        divya(root.right,s*10+root.val);
    }
    public int sumNumbers(TreeNode root) {
        sum=0;
        divya(root,0);
        return sum;
    }
}